packages/compartment-mapper/src/import-hook.js

```diff
@@ -108,12 +108,21 @@ const findRedirect = ({
   compartmentDescriptors,
   compartments,
   absoluteModuleSpecifier,
-  packageLocation,
 }) => {
-  const moduleSpecifierLocation = new URL(
-    absoluteModuleSpecifier,
-    packageLocation,
-  ).href;
+  /**
+   * Computes the relative path to a module from its compartment location (including a leading `./`)
+   *
+   * @param {string} moduleSpecifierLocation Absolute path (not URL) to a module specifier
+   * @param {string} location Compartment location
+   * @returns {string} Redirect static module interface; the `specifier` prop of this value _must_ be a relative path with leading `./`
+   */
+  const relativeSpecifier = (moduleSpecifierLocation, location) => {
+    const relativePath = moduleSpecifierLocation.replace(location, '');
+    return relativePath.startsWith('.') ? relativePath : `./${relativePath}`;
```

Comment:
how could a result of `.replace(location, '')` ever produce a leading dot?

Reply:
hm, right. I think I was worried about somehow getting a `../` somewhere. but I didn't think it through.